styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -126,13 +123,19 @@ private ExecutionDescription getExecDescription(WorkflowInstance workflowInstanc
           workflowId, workflowInstance, errors));
     }
 
+    final Map<String, String> env = new HashMap<>(workflow.configuration().env());
+    data.triggerParameters().ifPresent(p -> env.putAll(p.env()));
+
+    final Optional<String> dockerImage = workflow.configuration().dockerImage();
     final List<String> dockerArgs = workflow.configuration().dockerArgs()
         .orElse(Collections.emptyList());
     final List<String> command = argsReplace(dockerArgs, workflowInstance.parameter());
+    final Optional<FlyteExecConf> flyteExecConf = workflow.configuration().flyteExecConf();
 
-    final Map<String, String> env = new HashMap<>(workflow.configuration().env());
-    data.triggerParameters().ifPresent(p -> env.putAll(p.env()));
-
+    if (workflow.configuration().flyteExecConf().isEmpty() && dockerImage.isEmpty()) {
+      throw new MissingRequiredPropertyException(format("%s has no execution configuration, "
+                                                        + "halting %s", workflowId, workflowInstance));
+    }
```

Comment:
The workflow configuration shouldn’t be able to have both flyteExec and dockerImage, right? I started thinking about what would happen in that case for the runner handlers. Should we add a check to verify this somewhere? @narape

Reply:
nvm, validateWorkflow does this above